app/dns-bind.hoon

```diff
@@ -0,0 +1,895 @@
+/-  *dns-bind, dns, hall
+/+  tapp, stdio
+::
+::  tapp types and boilerplate
+::
+=>  |%
+    ++  collector-app  `dock`[~zod %dns-collector]
+    +$  app-state
+      $:  %0
+          :: nem: authoritative state
+          ::
+          nem=(unit nameserver)
+      ==
+    +$  peek-data  _!!
+    +$  in-poke-data
+      $%  [%dns-authority =authority]
+          [%dns-bind =ship =target]
+          [%handle-http-request =inbound-request:http-server]
+      ==
+    +$  out-poke-data
+      $%  [%dns-bind =ship =target]
+          [%dns-complete =ship =binding:dns]
+          [%drum-unlink =dock]
+      ==
+    +$  in-peer-data
+      $%  [%dns-request =request:dns]
+      ==
+    +$  out-peer-data  ~
+    ++  tapp
+      %:  ^tapp
+        app-state
+        peek-data
+        in-poke-data
+        out-poke-data
+        in-peer-data
+        out-peer-data
+      ==
+    ++  tapp-async  tapp-async:tapp
+    ++  stdio  (^stdio out-poke-data out-peer-data)
+    --
+::
+::  oauth2 implementation
+::
+=>  |%
+    ::  +oauth2-config: as one would expect
+    ::
+    +$  oauth2-config
+      $:  auth-url=@t
+          exchange-url=@t
+          domain=turf
+          initial-path=path
+          redirect-path=path
+          scopes=(list @t)
+      ==
+    ::  +oauth2: library core
+    ::
+    ++  oauth2
+        |_  [our=@p now=@da config=oauth2-config code=@t =hart:eyre secrets=@t]
+      ::
+      ++  local-uri
+        |=  [our=ship =path]
+        ^-  @t
+        ::  XX can't scry in +mule
+        ::
+        :: =/  =hart:eyre  .^(hart:eyre %r /(scot %p our)/host/real)
+        (crip (en-purl:html [hart [~ path] ~]))
+      ::
+      ::  XX can't scry in +mule
```

Comment:
`+mule` calls `+mute`, which calls `+mock` with a trivial scry handler.  Would it be possible to call `+mock` more directly and supply a scry handler that calls `.^`?

Reply:
We could do that, but it's kind of like a double-sided chainsaw from a determinism standpoint. I assume that's why it's not already in the stdlib. To support scry via, we'll either need to do that or operate in a vase context as %gall does.